scripts/lint.py

```diff
@@ -748,6 +749,23 @@ def lint_file(filepath: str, args: Any) -> int:
     return num_errors
 
 
+def lint_crate_docs(should_ignore: Callable[[Any], bool]) -> int:
+    """Make sure ARCHITECTURE.md talks about every single create we have."""
+
+    crates_dir = Path(__file__).parent.parent / "crates"
```

Comment:
This isn't very robust, considering this very PR makes me want to split `crates` into multiple sub-directories.

I suggest you find the list of crates in some other way, e,.g. folder names containing `Cargo.toml` (ignoring the root/workspace).

Reply:
Done